text/0000-export-to-oci.md

```diff
@@ -0,0 +1,707 @@
+  - This feature will help to unblock uses cases like
+    - OCI annotations. See [RFC](https://github.com/buildpacks/rfcs/pull/196)
+    - Cosign integration. See [RFC](https://github.com/buildpacks/rfcs/pull/195)
+    - Export to tarball. See [issue](https://github.com/buildpacks/lifecycle/issues/423)
+
+### What is the expected outcome?
+
+Lifecycle will be capable of exporting the application image into disk in [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format. The image saved on disk could have the following considerations:
+
+- The `blobs` directory MAY be missing `base image` or `run image` blobs. These layers may not be needed on disk as they could be already accessible in a blob store.
+- The `blobs` directory SHOULD always have buildpacks generated `blobs`.
+
+
+# What it is
+[what-it-is]: #what-it-is
+
+The proposal is to add a new capability to the lifecycle (enabled by configuration) to resolve any **image reference** (input or output) to a disk location in [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format. It means, instead of interacting with a daemon or registry lifecycle will interact against the filesystem to read or write any **image reference**.
+
+The target personas affected by this change are:
+
+- **Platform implementors**: they will have to take care of the responsibility of creating a store resource on disk in [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format and pass it through the lifecycle during the phases execution.  
+
+The process of writing any image on disk in [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format could be expensive in terms of hard drive space or IO operation (compressing or uncompressing layers). In order to provide flexibility for the implementation, the `analyzer` or `exporter` binaries only require the *Image Manifest* and the *Image Config* to execute their operations on the previous image and run image; based on this, we proposed the Lifecycle can be configured to work with a partial representation of the images on disk, meaning that some blobs MAY be missing (which is ok according to the [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format). The missing blobs SHOULD be those that are already available in a daemon or registry.  
+
+Let's see some examples of the proposed behavior
+
+## Examples
+
+### Requirements
+
+A folder on disk (accessible by the lifecycle) is required to execute the feature, this new folder works as a local registry and the content must be in [OCI Image Layout](https://github.com/opencontainers/image-spec/blob/main/image-layout.md) format.
```

Comment:
Perhaps worth noting, at least in the current implementation, the media type in index.json must be `application/vnd.docker.distribution.manifest.v2+json` or `application/vnd.oci.image.index.v1+json` (`skopeo copy -a docker://busybox:latest <dest>` will give you the latter, whereas `crane pull busybox:latest <dest> --format=oci` will give you `application/vnd.docker.distribution.manifest.list.v2+json` which will fail). This is at least worth calling out in our documentation in case users try to piece together inputs from their own tooling.

Reply:
I think this is a valuable comment when documenting the feature. I will add into the doc issue to do not forget about it